backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java

```diff
@@ -460,7 +493,10 @@ public boolean addBreadcrumb(String message, BacktraceBreadcrumbType type, Backt
      * @return true if the breadcrumb was successfully added
      */
     public boolean addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type) {
-        return database.getBreadcrumbs().addBreadcrumb(message, attributes, type);
+        if (isBreadcrumbsEnabled()) {
```

Comment:
small nit:
it would be better if we return without nesting more code in the if statement. you can save one tab and overall code would be easier to read from the top to the bottom if you inverse the if statement
```
if (!isBreadcrumbsEnabled()) {
return false;
} 
return database.getBreadcrumbs().addBreadcrumb(message, attributes, type);
```